// @seanhouli/react-mapbox-search/dist/Components/SearchInput.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _search = _interopRequireDefault(require("react-icons/lib/fa/search"));

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  height: 100%;\n  width: 100%;\n  text-indent: 40px;\n  font-size: 15px;\n  border: 1px solid #d3d3d3;\n  border-radius: 3px;\n  ", "\n  ", "\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  width: 100%;\n  height: 100%;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var Wrapper = _styledComponents["default"].div(_templateObject());

var InputWrapper = _styledComponents["default"].input(_templateObject2(), function (props) {
  return props.hasResults && "border-bottom-left-radius: 0;";
}, function (props) {
  return props.hasResults && "border-bottom-right-radius: 0;";
});

var SearchInput = function SearchInput(_ref) {
  var value = _ref.value,
      handleInputChange = _ref.handleInputChange,
      hasResults = _ref.hasResults,
      handleFocus = _ref.handleFocus,
      handleBlur = _ref.handleBlur,
      searchHint = _ref.searchHint;
  return _react["default"].createElement(Wrapper, null, _react["default"].createElement(_search["default"], null), _react["default"].createElement(InputWrapper, {
    displayName: "input",
    hasResults: hasResults,
    onFocus: handleFocus,
    onBlur: handleBlur,
    placeholder: searchHint,
    value: value,
    onChange: handleInputChange
  }));
};

var _default = SearchInput;
exports["default"] = _default;
