// @seanhouli/react-mapbox-search/dist/Components/Suggestion.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  font-size: 14px;\n  padding: 1rem 1.5rem 1rem 1.5rem;\n  cursor: pointer;\n  width: 100%;\n\n  &:hover {\n    color: white;\n    background-color: ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var Wrapper = _styledComponents["default"].div(_templateObject(), function (props) {
  return props.selectColor;
});

var Suggestion = function Suggestion(_ref) {
  var place = _ref.place,
      clickHandler = _ref.clickHandler,
      cursorIdx = _ref.cursorIdx,
      idx = _ref.idx,
      selectColor = _ref.selectColor,
      mouseInSuggestions = _ref.mouseInSuggestions,
      isTouch = _ref.isTouch;
  return _react["default"].createElement(Wrapper, {
    selectColor: selectColor,
    style: cursorIdx === idx && !mouseInSuggestions && !isTouch ? {
      color: "white",
      background: selectColor
    } : null,
    onMouseDown: function onMouseDown(event) {
      return clickHandler({
        location: place,
        event: event
      });
    }
  }, place.place_name);
};

var _default = Suggestion;
exports["default"] = _default;
